pkg/resource/parameter_group/post_set_output.go

```diff
@@ -0,0 +1,118 @@
+
+package parameter_group
+
+import (
+	"context"
+	svcapitypes "github.com/aws-controllers-k8s/memorydb-controller/apis/v1alpha1"
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+	svcsdk "github.com/aws/aws-sdk-go/service/memorydb"
+)
+
+func (rm *resourceManager) setParameters(
+	ctx context.Context,
+	ko *svcapitypes.ParameterGroup,
+) (*svcapitypes.ParameterGroup, error) {
+	if ko.Spec.Name == nil {
+		return ko, nil
+	}
+
+	parameters, err := rm.describeParameters(ctx, ko.Spec.Name)
+	if err != nil {
+		return nil, err
+	}
+	parameterNameValues := []*svcapitypes.ParameterNameValue{}
+	for _, p := range parameters {
+		sp := svcapitypes.ParameterNameValue{
+			ParameterName:  p.Name,
+			ParameterValue: p.Value,
+		}
+		parameterNameValues = append(parameterNameValues, &sp)
+	}
+	ko.Spec.ParameterNameValues = parameterNameValues
+	// Update parameter value to nil to not duplicate status and spec values.
+	for _, p := range parameters {
+		p.Value = nil
+	}
+	ko.Status.Parameters = parameters
+	return ko, nil
+}
+
+func (rm *resourceManager) describeParameters(
+	ctx context.Context,
+	parameterGroupName *string,
+) ([]*svcapitypes.Parameter, error) {
+
+	parameters := []*svcapitypes.Parameter{}
+	var nextToken *string = nil
+	for {
+		input, err := rm.newDescribeParametersRequestPayload(parameterGroupName, nextToken)
+		if err != nil {
+			return nil, err
+		}
+		response, respErr := rm.sdkapi.DescribeParametersWithContext(ctx, input)
+		rm.metrics.RecordAPICall("GET", "DescribeParameters", respErr)
+		if respErr != nil {
+			if awsErr, ok := ackerr.AWSError(respErr); ok && awsErr.Code() == "ParameterGroupNotFound" {
+				return nil, ackerr.NotFound
+			}
+			rm.log.V(1).Info("Error during DescribeParameters", "error", respErr)
+			return nil, respErr
+		}
+		if response.Parameters == nil {
+			break
+		}
+
+		// Update the next token
+		nextToken = response.NextToken
+
+		for _, p := range response.Parameters {
+			sp := svcapitypes.Parameter{
+				Name:                 p.Name,
+				Value:                p.Value,
+				Description:          p.Description,
+				DataType:             p.DataType,
+				AllowedValues:        p.AllowedValues,
+				MinimumEngineVersion: p.MinimumEngineVersion,
+			}
+			parameters = append(parameters, &sp)
+		}
+
+		if nextToken == nil || *nextToken == ""{
+			break
+		}
+	}
+
+	return parameters, nil
+}
+
+// newDescribeParametersRequestPayload returns SDK-specific struct for the HTTP request
+// payload of the DescribeParameters API to get properties that have
+// given parameterGroupName and given source
+func (rm *resourceManager) newDescribeParametersRequestPayload(
+	parameterGroupName *string,
+	nextToken *string,
+) (*svcsdk.DescribeParametersInput, error) {
+	res := &svcsdk.DescribeParametersInput{}
+
+	if parameterGroupName != nil {
+		res.SetParameterGroupName(*parameterGroupName)
+	}
+
+	if nextToken != nil {
+		res.SetNextToken(*nextToken)
+	}
+
+	return res, nil
+}
```

Comment:
nit: This function is not needed, you can build the request struct directly when calling `rm.sdkapi.DescribeParametersWithContext` in L64
e.g:

```go
rm.sdkapi.DescribeParametersWithContext(ctx, &svcsdk.DescribeParametersInput{
    ParameterGroupName: parameterGroupName,
    NextToken: nil,
})
```